File: 23/viejo/23.py
#!/usr/bin/python

def sumaDivisores(numero):
    a=0
    for n in range(1,numero):
        if numero%n==0: a+=n
    return a


enteros=[]
abundantes=[]
deficientes=[]
perfectos=[]

for entero in range(1,28123):
    if ( entero > 30 and entero%2==0) : continue
    enteros.append(entero)

for a1 in range(1,28123):
    if a1>sumaDivisores(a1): continue
    for a2 in range(1,a1+1):
        if a2>sumaDivisores(a1): continue

        a3=a1+a2
        if a3 in enteros:
            #print a1,a2,a3
            enteros.remove(a3)

    print len(enteros)

print enteros

suma=0
for a in enteros:
    suma+=a

print a

File: 23/viejo/23.c
#include <stdio.h>

int sumaDivisores(int numero) {
    int suma=0;
    unsigned int i;

    for (i=1;i<numero;i++) {
        if (numero%i ==0) {
            suma+=i;
        }
    }

    return suma;
}

void main() {

    unsigned int i,a1,a2;
    short salida;
    long suma=0;

    for (i=1;i<28123;i++) {
        if ( i%2==0 && i>30 ) { i++; }; // 24 + 6
        salida=0;

        for (a1=1; a1<i;a1++) { 
            if ( sumaDivisores(a1) > a1 ) { //a1 es abundante
                for (a2=2;a2<=a1;a2++) {
                    if ( sumaDivisores(a2) > a2 ) { //a2 es abundante
                        if ( a1+a2==i ) { 
                            printf("%d=%d+%d\n",i,a1,a2);
                            salida=1; break; }
                    }   
                
                if ( salida == 1) { break; }
                }   
            }
        }
        if ( salida == 0) { suma+=i; printf("%d\n",i);}
    }

    printf("!!!!!!!!!!!!!!!!!\n%d\n",suma);
}
